### PR TITLE
adding cron job to products

### PR DESCRIPTION
**Pull request by Y-elv**

### CODE REVIEW

**Review by hbapte — DISMISSED** [review]

**Review by Fabrice-Dush — DISMISSED** [review]

File: src/services/cronJob.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import cron from "node-cron";
+import updateExpiredProducts from "../helpers/updateExpiredProducts";
+import { Request, Response } from "express";
+
+const cronSchedule = "* * * * *";
```
Comment (AimePazzo) [inline src/services/cronJob.ts:5]: why do you use environment variables so we can modify this variable `cronSchedule`

File: src/helpers/updateExpiredProducts.ts [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+          await sendEmail(
+            user.email,
+            "Product Expired",
+            `Dear ${user.firstName}, your product ${product.name} has expired and now is unavailable.`
+          );
+        }
+      }
+    }
+  } catch (error) {
+    console.error("Error updating expired products:", error.message);
```
Comment (SaddockAime) [inline src/helpers/updateExpiredProducts.ts:67]: Remove this console

File: src/helpers/updateExpiredProducts.ts [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    return res.status(500).json({ error: "Internal server error" });
```
Comment (SaddockAime) [inline src/helpers/updateExpiredProducts.ts]: Use httpStatus like this:
res.status(httpStatus.INTERNAL_SERVER_ERROR).json({
      status: httpStatus.INTERNAL_SERVER_ERROR,
      error: error.message,
    });

File: src/services/cronJob.ts [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+
+export const startCronJobMiddleware = () => {
+  cron.schedule(cronSchedule, async () => {
+    try {
+      const req = {} as Request;
+      const res = {} as Response;
+      console.log("am cronning all products...");
```
Comment (SaddockAime) [inline src/services/cronJob.ts]: Remove this console

File: src/services/cronJob.ts [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+      await updateExpiredProducts(req, res);
+    } catch (error) {
+      console.error("Error updating expired products:", error);
```
Comment (SaddockAime) [inline src/services/cronJob.ts]: use httpStatus for responses:
res.status(httpStatus.INTERNAL_SERVER_ERROR).json({
      status: httpStatus.INTERNAL_SERVER_ERROR,
      error: error.message,
    });

**Review by ProgrammerDATCH — DISMISSED** [review]

**Review by ndahimana154 — DISMISSED** [review]

**Review by MANISHIMWESalton — DISMISSED** [review]

**Review by ProgrammerDATCH — DISMISSED** [review]

**Pull request merged**